main.go

```diff
@@ -4,39 +4,207 @@ import (
 	"bytes"
 	"context"
 	"encoding/json"
-	"flag"
+	"errors"
 	"fmt"
 	"io"
 	"log"
 	"os"
+	"path/filepath"
+	"regexp"
 	"runtime"
+	"strings"
+	"time"
 
-	sensu "github.com/sensu/sensu-go/api/core/v2"
+	"github.com/sensu/sensu-go/types"
+	"github.com/sensu/sensu-plugin-sdk/sensu"
 )
 
+// Config represents the check plugin config.
+type Config struct {
+	sensu.PluginConfig
+	LogFile           string
+	LogFileExpr       string
+	LogPath           string
+	StateDir          string
+	Procs             int
+	MatchExpr         string
+	MatchStatus       int
+	InverseMatch      bool
+	MaxBytes          int64
+	EventsAPI         string
+	IgnoreInitialRun  bool
+	DisableEvent      bool
+	DryRun            bool
+	Verbose           bool
+	EnableStateReset  bool
+	CheckNameTemplate string
+}
+
 var (
-	logFile          = flag.String("log", "", "path to log file (required)")
-	procs            = flag.Int("procs", runtime.NumCPU(), "number of parallel analyzer processes")
-	match            = flag.String("match", "", "RE2 regexp matcher expression (required)")
-	stateFile        = flag.String("state", "", "state file for incremental log analysis (required)")
-	eventStatus      = flag.Int("event-status", 1, "event status on positive match")
-	eventsAPI        = flag.String("api-url", "http://localhost:3031/events", "agent events API URL")
-	maxBytes         = flag.Int64("max-bytes", 0, "max number of bytes to read (0 means unlimited)")
-	ignoreInitialRun = flag.Bool("ignore-initial-run", false, "suppresses alerts for any matches found on the first run of the plugin")
-)
+	useStdin            = false
+	defaultNameTemplate = "{{ .Check.Name }}-alert"
+	logs                = []string{}
+	plugin              = Config{
+		PluginConfig: sensu.PluginConfig{
+			Name:     "sensu-check-log",
+			Short:    "Check Log",
+			Keyspace: "sensu.io/plugins/sensu-check-log/config",
+		},
+	}
 
-const (
-	// StatusOK represents a 0 exit status
-	StatusOK = 0
-	// StatusWarn represents a 1 exit status
-	StatusWarn = 1
-	// StatusCrit represents a 2 exit status
-	StatusCrit = 2
+	options = []*sensu.PluginConfigOption{
+		&sensu.PluginConfigOption{
+			Path:      "log-file",
+			Env:       "CHECK_LOG_FILE",
+			Argument:  "log-file",
+			Shorthand: "f",
+			Default:   "",
+			Usage:     "Log file to check. (Required if --log-file-expr not used)",
+			Value:     &plugin.LogFile,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "log-file-expr",
+			Env:       "CHECK_LOG_FILE_EXPR",
+			Argument:  "log-file-expr",
+			Shorthand: "e",
+			Default:   "",
+			Usage:     "Log file regexp to check. (Required if --log-file not used)",
+			Value:     &plugin.LogFileExpr,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "log-path",
+			Env:       "CHECK_LOG_PATH",
+			Argument:  "log-path",
+			Shorthand: "p",
+			Default:   "",
+			Usage:     "Log path for basis of log file regexp. (Required if --log-file-expr used)",
```

Comment:
Required if --log-file-expr _not_ used, right?

Reply:
No required if --log-file-expr used.. set the file system base directory from which the rexpr will walk the directory tree.

log-path  "/var/log"  for example  would restrict your regexp to just walk the /var/log/ directory tree

Reply:
would setting a default like /var/log  make more sense?

Reply:
I've set the default for --log-path to /var/log  and updated textual description for the option to be a little clearer.